src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.hide();
+      animal.health -= 50;
+
+      if (animal.health <=0) this.die();
```

Comment:
[CODE LOGIC] - The `bite` method should not call the `hide` method on the animal being bitten. According to the task description, the `hide` method should only set the `hidden` property to `true`, and it should not be toggled or called by other methods like `bite`. Remove the call to `hide` within the `bite` method.